src/lib.rs

```diff
@@ -181,3 +183,63 @@ impl<T> From<T> for SyncWrapper<T> {
         Self::new(value)
     }
 }
+
+
+pub struct SyncFuture<F> {
+    inner: SyncWrapper<F>
+}
+impl <F: Future> SyncFuture<F> {
+    /// Create a `Future` which is `Sync`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sync_wrapper::{SyncWrapper, SyncFuture};
+    ///
+    /// let fut = async { 1 };
+    /// let fut = SyncFuture::new(fut);
+    /// ```
+    pub fn new(inner: F) -> Self {
+        Self { inner: SyncWrapper::new(inner) }
+    }
+    pub fn into_inner(self) -> SyncWrapper<F> {
+        self.inner
```

Comment:
Since SyncWrapper is an implementation detail of SyncFuture, I’d want to return plain `F` from this function.